slasher/rpc/detect_update_min_max_span.go

```diff
@@ -0,0 +1,142 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// Detector is an interface used to implement the slashable surrounding/surrounded
+// vote detection methods.
+type Detector interface {
+	Detect(attestationEpochSpan uint64, recorderEpochSpan *ethpb.MinMaxEpochSpan, sourceEpoch uint64) uint64
```

Comment:
Instead of defining an interface, can you define a function type? this is a bit cumbersome to use. 

```go
type DetectFn = func(attestationEpochSpan uint64, recorderEpochSpan *ethpb.MinMaxEpochSpan, sourceEpoch uint64) uint64
```

Then you can change the API of detectSlashingByEpochSpan
```go
func detectSlashingByEpochSpan(source, target, validatorIdx uint64, detect DetectFn) {
  if detect(...) > 0 {
     // do something
   }
}
```